src/app/organization/offices/edit-office/edit-office.component.ts

```diff
@@ -0,0 +1,82 @@
+import { Component, OnInit } from '@angular/core';
+import { FormGroup, FormBuilder, Validators } from '@angular/forms';
+import { OrganizationService } from 'app/organization/organization.service';
+import { ActivatedRoute, Router } from '@angular/router';
+import { DatePipe } from '@angular/common';
+
+@Component({
+  selector: 'mifosx-edit-office',
+  templateUrl: './edit-office.component.html',
+  styleUrls: ['./edit-office.component.scss']
+})
+export class EditOfficeComponent implements OnInit {
+
+  /** Selected Data. */
+  officeData: any;
+  /** Office form. */
+  officeForm: FormGroup;
+  /** Date. */
+  openedOn: any;
+
+    /**
+     * Retrieves the charge data from `resolve`.
+     * @param {ProductsService} organizationService Organization Service.
+     * @param {FormBuilder} formBuilder Form Builder.
+     * @param {ActivatedRoute} route Activated Route.
+     * @param {Router} router Router for navigation.
+     * @param {MatDialog} dialog Dialog reference.
+     * @param {DatePipe} datepipe Convert Date.
+     */
+
+    constructor(
+      private organizationService: OrganizationService,
+      private formBuilder: FormBuilder,
+      private route: ActivatedRoute,
+      private router: Router,
+      private datepipe: DatePipe
+    ) {
+      this.route.data.subscribe((data: { officeTemplate: any }) => {
+        this.officeData = data.officeTemplate;
+        const date_string = this.officeData.openingDate[0].toString() + '-' + this.officeData.openingDate[1].toString() + '-' + this.officeData.openingDate[2].toString();
```

Comment:
no need for date_string variable directly assign `this.openedOn = '${this.officeData.openingDate[0]}-${this.officeData.openingDate[1]}-${this.officeData.openingDate[2]}`'